src/freertos_drivers/spiffs/SPIFFS.cxx

```diff
@@ -45,6 +46,20 @@
 #define O_DIRECT _FDIRECT
 #endif
 
+/// Provide mutex lock.
+/// @param fs reference to the file system instance
+void SPIFFS::extern_lock(struct spiffs_t *fs)
```

Comment:
These "extern_lock" functions seem to be declared "inline" in the header file.  Don't these have to be implemented as inline in the header file as well?

Reply:
Technically their definition must be available in the compilation unit where they are used. That's only in SPIFFS.cxx for this case.
The assembly code generated is as follows:
```
0803f5e0 <extern_spiffs_lock>:
 803f5e0:	b508      	push	{r3, lr}
 803f5e2:	6f80      	ldr	r0, [r0, #120]	; 0x78
 803f5e4:	3024      	adds	r0, #36	; 0x24
 803f5e6:	f7d2 facf 	bl	8011b88 <OSMutex::lock()>
 803f5ea:	bd08      	pop	{r3, pc}
```
which shows that inlining happened correctly.